File: 125-valid-palindrome/valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    
    let cleaned  = s.toLowerCase().replace(/[^a-z0-9]/g, '');
    let isPalindrome = true;

    let start = 0;
    let end = cleaned.length-1;

    while(start<end){
        if(cleaned[start]==cleaned[end]){
            start++;
            end--;

        }

        else{
            return isPalindrome = false;

        }

        
    }

   if(isPalindrome==true){
    return true;
   }

   else{
    return false;
   }

    
};

File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {

        
        int start = 0;
        int end = s.length()-1;
        Boolean isPalindrome = true;

        while(start<end){
            if(!Character.isLetterOrDigit(s.charAt(start))){
                start++;
            }

            else if(!Character.isLetterOrDigit(s.charAt(end))){
                end--;
            }

            else{
                if(Character.toLowerCase(s.charAt(start))==Character.toLowerCase(s.charAt(end))){
                    start++;
                    end--;
                }

                else{
                    return isPalindrome = false;
                }
            }

            
        }

        if(isPalindrome==true){
            return true;
        }

        else{
            return false;
        }
        
    }
}